lib/numericSequence.fz

```diff
@@ -0,0 +1,83 @@
+# This file is part of the Fuzion language implementation.
+#
+# The Fuzion language implementation is free software: you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as published
+# by the Free Software Foundation, version 3 of the License.
+#
+# The Fuzion language implementation is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
+# License for more details.
+#
+# You should have received a copy of the GNU General Public License along with The
+# Fuzion language implementation.  If not, see <https://www.gnu.org/licenses/>.
+
+
+# -----------------------------------------------------------------------
+#
+#  Tokiwa Software GmbH, Germany
+#
+#  Source code of Fuzion standard library feature numericSequence
+#
+#  Author: Michael Lill (michael.lill@tokiwa.software)
+#
+# -----------------------------------------------------------------------
+
+# numericSequence -- a Sequence whose elements inherit from numeric
+#
+numericSequence<N: numeric<N>>(from Sequence<N>) : Sequence<N> is
+
+  # create a list from this Sequence.
+  #
+  redef asList => from.asList
+
+
+  # the arithmetic mean of the sequence
+  # https://en.wikipedia.org/wiki/Arithmetic_mean
+  average option<N> is
```

Comment:
I try to separate the comment from the feature by a blank comment line
```
  #
```
even though I am not fully consistent with this. Try to do this as well.